pkgs/by-name/gr/granian/package.nix

```
@@ -0,0 +1,46 @@
+{
+  lib,
+  python3,
+  fetchFromGitHub,
+  rustPlatform,
+  python312Packages,
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "granian";
+  version = "1.5.2";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "emmett-framework";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-3kwdPLOkKKSY1w42av3ejarrRKNtai8KB80ZrujSjPo=";
+  };
+
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "pyo3-log-0.11.0" = "sha256-UU8064vM7cf20lwfifyPC205CY4tIgif0slDz/Pjf9Q=";
+    };
+  };
+
+  nativeBuildInputs = with rustPlatform; [
+    cargoSetupHook
+    maturinBuildHook
+  ];
+
+  dependencies = [
+    python312Packages.uvloop
+    python312Packages.click
+  ];
+
+  meta = {
+    description = "A Rust HTTP server for Python ASGI/WSGI/RSGI applications.";
```

```suggestion
    description = "Rust HTTP server for Python ASGI/WSGI/RSGI applications";
```

no leading article, no trailing period, per guidelines